lib/facter/env_temp_variable.rb

```diff
@@ -0,0 +1,3 @@
+Facter.add(:env_temp_variable) do
+  setcode { ENV['TEMP'] }
```

Comment:
There are cases where TEMP is not defined, like when using cygwin or when building the system for the first time. The latter doesn't apply, but the first might, so we should have a backup. Plus MSI will barf if the folder doesn't exist (I just dealt with this). So default the folder if the value is null, ensure that it is forward slashes, and ensure that the folder exists. I see ensuring the folder exists as a custom function or a file directive.